src/Microsoft.ML.Core/Data/RoleMappedSchema.cs

```diff
@@ -256,42 +193,40 @@ private RoleMappedSchema(Schema schema, Dictionary<string, IReadOnlyList<ColumnI
             }
         }
 
-        private RoleMappedSchema(Schema schema, Dictionary<string, List<ColumnInfo>> map)
+        private RoleMappedSchema(Schema schema, Dictionary<string, List<Schema.Column>> map)
             : this(schema, Copy(map))
         {
         }
 
-        private static void Add(Dictionary<string, List<ColumnInfo>> map, ColumnRole role, ColumnInfo info)
+        private static void Add(Dictionary<string, List<Schema.Column>> map, ColumnRole role, Schema.Column column)
         {
             Contracts.AssertValue(map);
             Contracts.AssertNonEmpty(role.Value);
-            Contracts.AssertValue(info);
 
             if (!map.TryGetValue(role.Value, out var list))
             {
-                list = new List<ColumnInfo>();
+                list = new List<Schema.Column>();
                 map.Add(role.Value, list);
             }
-            list.Add(info);
+            list.Add(column);
         }
 
-        private static Dictionary<string, List<ColumnInfo>> MapFromNames(Schema schema, IEnumerable<KeyValuePair<ColumnRole, string>> roles, bool opt = false)
+        private static Dictionary<string, List<Schema.Column>> MapFromNames(Schema schema, IEnumerable<KeyValuePair<ColumnRole, string>> roles, bool opt = false)
         {
             Contracts.AssertValue(schema);
             Contracts.AssertValue(roles);
 
-            var map = new Dictionary<string, List<ColumnInfo>>();
+            var map = new Dictionary<string, List<Schema.Column>>();
             foreach (var kvp in roles)
             {
                 Contracts.AssertNonEmpty(kvp.Key.Value);
                 if (string.IsNullOrEmpty(kvp.Value))
                     continue;
-                ColumnInfo info;
-                if (!opt)
-                    info = ColumnInfo.CreateFromName(schema, kvp.Value, kvp.Key.Value);
-                else if (!ColumnInfo.TryCreateFromName(schema, kvp.Value, out info))
-                    continue;
-                Add(map, kvp.Key.Value, info);
+                var info = schema.GetColumnOrNull(kvp.Value);
+                if (info.HasValue)
+                    Add(map, kvp.Key.Value, info.Value);
+                else if (!opt)
+                    throw Contracts.ExceptParam(nameof(schema), $"{kvp.Value} column '{kvp.Key.Value}' not found");
```

Comment:
>ExceptParam [](start = 36, length = 11)

do you think `ExceptSchemaMismatch` is a good fit here?

Reply:
Maybe. TBH I was never quite sure what `ExceptSchemaMismatch` was for exactly.